tests/system/common.py

```diff
@@ -819,3 +822,26 @@ def kill_process_on_host(hostname, pattern):
     else:
         print("Killed no pids")
     return pids
+
+
+def service_available_predicate(service_name):
+    url = dcos_service_url(service_name)
+    try:
+        response = http.get(url)
+        return response.status_code == 200
+    except Exception as e:
+        return False
+
+
+# since exhibitor might take ~ 20 minutes to start up, we need to wait for it
+def wait_for_service_endpoint(service_name, timeout_sec=120):
+    """Checks the service url if available it returns true, on expiration
+    it returns false"""
+    @retrying.retry(wait_fixed=1000, stop_max_attempt_number=1200, retry_on_exception=ignore_exception)
```

Comment:
do we really want to wait 20 minutes for exhibitor? What kind of software starts up 20 minutes? That's crazy...

Shouldn't we maybe give up earlier but fail with message that would clearly state that we failed waiting on exhibitor?

Btw. the usage of exhibitor in shakedown is kind of questionable too. Waiting 20 minutes for current number of masters... can't we get that information from somewhere else?

Reply:
> do we really want to wait 20 minutes for exhibitor? What kind of software starts up 20 minutes? That's crazy...

**If** the exhibitor is responding. If not (ex. other cluster issues) and it takes 120 seconds to time-out, it's going to take 40 hours 😄

Reply:
Therefore let's not include `TimeoutExpired` exception in the ignored ones. A connection error (in my opinion) should be considered critical and the test should fail.

Reply:
@alenkacz 
> do we really want to wait 20 minutes for exhibitor?

That's what I see in adminrouter logs. But the exhibitor starts up eventually.

> Shouldn't we maybe give up earlier 

Then we have to live with flaky tests. (and this is what we do right now).
> can't we get that information from somewhere else?

Since data about masters is stored in zookeeper, I'm not sure if we can avoid querying exhibitor.

@wavesoft 
> it's going to take 40 hours 😄

Not really since the http timeout is 5 seconds. But I'll adjust it to match 20 minutes, thanks
https://github.com/dcos/dcos-cli/blob/887f490b34ab6d8e572c189ef708e326a2d96df1/python/lib/dcos/dcos/http.py#L297

Reply:
I still doubt the 20 minute time.
- it sounds ridiculous that it boots up that long (in integration tests, so not under load or with a lots of data)
- next time we will see it boot upt for 40 minutes so we will increase?

The number just looks too long and kind of artificial (if it"s based on one log).

But I am open to counter-arguments :)